src/components/Sidebar.svelte

```diff
@@ -1,45 +1,38 @@
 <script lang="ts">
-    import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+    import { createEventDispatcher} from 'svelte';
     export let sidebarOpen: boolean;
-    export let toggleButton;
+    export let toggleButton: HTMLElement | null;
     export let selectFranchise: (id: number) => void;
 
     const dispatch = createEventDispatcher();
-    let sidebar;
+    let sidebar: HTMLElement | null;
 
-    function handleSelectFranchise(id) {
+    function handleSelectFranchise(id: number) {
         selectFranchise(id);
+        dispatch('franchiseSelected', { id });
         sidebarOpen = !sidebarOpen;
     }
 
-    function handleClickOutside(event) {
-        if (!sidebar.contains(event.target) && !toggleButton.contains(event.target)) {
-            sidebarOpen = false;
-        }
-    }
-
-    function closeSidebar() {
+    function toggleSidebar() {
         sidebarOpen = !sidebarOpen;
     }
 
-    onMount(() => {
-        document.addEventListener('click', handleClickOutside);
-    });
-
-    onDestroy(() => {
-        document.removeEventListener('click', handleClickOutside);
-    });
+    function handleKeydown(event: KeyboardEvent) {
+        if (event.key === 'Escape' || event.key === 'Enter' || event.code === 'Space') {
+            toggleSidebar();
+        }
+    }
 </script>
 
 <div bind:this={sidebar} class="sidebar" class:open={sidebarOpen}>
     <div class="icons">
         <i class="fa-light fa-magnifying-glass nav-icon"></i>
         <i class="fa-light fa-shopping-cart nav-icon"></i>
         <i class="fal fa-user nav-icon"></i>
-        <i class="fal fa-xmark nav-icon" on:click={closeSidebar}></i>
+        <i bind:this={toggleButton} class="fal fa-xmark nav-icon" role="button" tabindex="-1" on:click={toggleSidebar} on:keydown={handleKeydown}></i>
     </div>
-    <a href="#" on:click={() => handleSelectFranchise(17)}>Star Wars</a>
-    <a href="#" on:click={() => handleSelectFranchise(4)}>Marvel</a>
+    <button on:click={() => handleSelectFranchise(17)}>Star Wars</button>
+    <button on:click={() => handleSelectFranchise(4)}>Marvel</button>
```

Comment:
```suggestion
    <button type="button" on:click={() => handleSelectFranchise(17)}>Star Wars</button>
    <button type="button" on:click={() => handleSelectFranchise(4)}>Marvel</button>
```

This is a good convention to keep in mind for a few reasons, one of them being that the `<button>` element acts as a `<input type="submit">` by default when used in a `<form>`. To prevent this, add the `type="button"` attribute.

Reply:
Thank you for the feedback i didn't know it worked like that i will keep it in mind while coding :

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-5fa8bc1e300230c2638d059cc9cd608ab21c79c6733942d5ddb7ca00bae85311